VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs

```diff
@@ -384,6 +385,12 @@ public async Task<ActionResult<OrderCreatedInfo>> CreateOrder([FromBody] BankCar
                 }
                 await Task.WhenAll(taskList.ToArray());
 
+                // Need to expire cache for resetting 'IsFirstTimeBuyer' after order creation
```

Comment:
This place for cache invalidation seems is quite ok, but I would like to prefer do not to know so much on this level. 
Alternative places for this logic are:
- in  an OrderPlacedEvent handler
- cache change token via polling requests
but again,  this place also seems good.

However, for the future for it will be better do not to have knowledge about contacts and orders at the security boundary level.  In the future, we need to define some abstraction for user data loaders (IUserExternalLoaders ) and load this data from other boundaries using this abstraction.

Reply:
Totally agree about tight coupling. Missed we have already have `OrderPlacedEvent` event, did not want to make new event only for this purpose.
Created handler.